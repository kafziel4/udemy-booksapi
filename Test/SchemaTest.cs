using BooksApi.GraphQL;
using BooksApi.Services;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace Test
{
    public class SchemaTest
    {
        [Fact]
        public async Task GenerateSchema_MatchesExistingSnapshot()
        {
            // Arrange
            var schema = await new ServiceCollection()
                .AddSingleton<Repository>()
                .AddGraphQLServer()
                .AddQueryType<Query>()
                .AddMutationType<Mutation>()
                .BuildSchemaAsync();

            // Act & Assert
            schema.ToString().MatchSnapshot();
        }
    }
}
