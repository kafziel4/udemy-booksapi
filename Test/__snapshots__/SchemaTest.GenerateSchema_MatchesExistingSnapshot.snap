schema {
  query: Query
  mutation: Mutation
}

type Author {
  id: UUID!
  name: String!
}

type AuthorPayload {
  record: Author
  error: String
}

type Book {
  id: UUID!
  title: String!
  author: Author!
}

type BookPayload {
  record: Book
  error: String
}

type Mutation {
  addAuthor(input: AuthorInput!): AuthorPayload!
  addBook(input: BookInput!): BookPayload!
}

type Query {
  books: [Book!]!
}

input AuthorInput {
  name: String!
}

input BookInput {
  title: String!
  author: UUID!
}

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

scalar UUID @specifiedBy(url: "https:\/\/tools.ietf.org\/html\/rfc4122")
